### PR TITLE
Aerospike state store implementation

### PR DESCRIPTION
**Pull request by abhirockzz**
# Description

State store implementation for Aerospike DB

## Issue reference

Issue #195 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

**Comment by abhirockzz** [comment]
FYI - extra files were added as a result of `make lint`

**Comment by yaron2** [comment]
@amanbha @youngbupark please review.

File: middleware/http/bearer/bearer_middleware.go [hunk 1]
```diff
@@ -55,17 +55,17 @@ func (m *Middleware) GetHandler(metadata middleware.Metadata) (func(h fasthttp.R
...
 			}
 			rawToken := authHeader[bearerPrefixLength:]
 			_, err := verifier.Verify(ctx, rawToken)
-			if (err != nil) {
+			if err != nil {
 				ctx.Error(fasthttp.StatusMessage(fasthttp.StatusUnauthorized), fasthttp.StatusUnauthorized)
 				return
 			}
-			
+
```
Comment (youngbupark) [inline middleware/http/bearer/bearer_middleware.go]: nit: Please create the pr only for formatting.

File: state/aerospike/aerospike.go [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+		return fmt.Errorf("aerospike: failed to connect %v", err)
+	}
+	aspike.client = c
+	aspike.namespace = metadata.Properties[namespace]
+	aspike.set = metadata.Properties[set]
+
+	return nil
+}
+
+//Set stores value for a key to Aerospike. It honors ETag (for concurrency) and consistency settings
```
Comment (youngbupark) [inline state/aerospike/aerospike.go]: nit: please add space after //

File: state/aerospike/aerospike.go [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+}
+
+func parseHosts(hostsMeta string) ([]*as.Host, error) {
+	hostPorts := []*as.Host{}
+	for _, hostPort := range strings.Split(hostsMeta, ",") {
+		if !strings.Contains(hostPort, ":") {
+			return nil, errInvalidHosts
+		}
+		host := strings.Split(hostPort, ":")[0]
+		port, _ := strconv.Atoi(strings.Split(hostPort, ":")[1])
```
Comment (youngbupark) [inline state/aerospike/aerospike.go]: I would use strconv.ParseUInt instead of Atoi for the safety.

File: state/cassandra/cassandra_test.go [hunk 4]
```diff
@@ -59,7 +59,6 @@ func TestGetCassandraMetadata(t *testing.T) {
 		assert.Equal(t, properties[username], metadata.username)
 		assert.Equal(t, properties[password], metadata.password)
 		assert.Equal(t, 9043, metadata.port)
-
```
Comment (youngbupark) [inline state/cassandra/cassandra_test.go]: nit: please create the pr for formatting separately.

**Comment by yaron2** [comment]
Hey @abhirockzz, can you address comments so we can merge? This is a great addition and we'd like to get it in..

File: state/aerospike/aerospike.go (hunk 3 above)
Comment (abhirockzz) [reply state/aerospike/aerospike.go]: done

File: state/aerospike/aerospike.go (hunk 2 above)
Comment (abhirockzz) [reply state/aerospike/aerospike.go]: done

**Comment by yaron2** [comment]
I think you deleted two unrelated files..

**Comment by abhirockzz** [comment]
> I think you deleted two unrelated files..

yes they were included as a result of `make lint` (as noted earlier)

**Comment by yaron2** [comment]
They were changed, which is OK.. your commit removed them entirely :)

Can you revert that commit?

**Comment by youngbupark** [comment]
> They were changed, which is OK.. your commit removed them entirely :)
> 
> Can you revert that commit?

General comment here: as long as we use `squash merge`, I suggest to make pr shorter and explicit even if it needs to additional PRs. It is a good practice to manage the commit history.

**Comment by youngbupark** [comment]
> > They were changed, which is OK.. your commit removed them entirely :)
> > Can you revert that commit?
> 
> General comment here: as long as we use `squash merge`, I suggest to make pr shorter and explicit even if it needs to additional PRs. It is a good practice to manage the commit history.

I am fine to merge this pr. Next please make pr shorter and explicit next time. 
@abhirockzz Thanks again for your contribution

**Comment by yaron2** [comment]
Thanks!

**Comment by abhirockzz** [comment]
> > They were changed, which is OK.. your commit removed them entirely :)
> > Can you revert that commit?
> 
> General comment here: as long as we use `squash merge`, I suggest to make pr shorter and explicit even if it needs to additional PRs. It is a good practice to manage the commit history.


> > > They were changed, which is OK.. your commit removed them entirely :)
> > > Can you revert that commit?
> > 
> > 
> > General comment here: as long as we use `squash merge`, I suggest to make pr shorter and explicit even if it needs to additional PRs. It is a good practice to manage the commit history.
> 
> I am fine to merge this pr. I do not want to ask reverting it again :) Please make pr shorter and explicit next time. :)
> 
> @abhirockzz Thanks again for your contribution


Agreed @youngbupark ! Thanks for your feedback and help

**Pull request merged**